src/snowflake/connector/result_set.py

```diff
@@ -155,7 +155,8 @@ def _fetch_pandas_all(self, **kwargs):
         if table:
             return table.to_pandas(**kwargs)
         else:
-            return pandas.DataFrame(columns=self.batches[0]._column_names)
+            column_names = [col.name for col in self.batches[0].schema]
```

Comment:
What do you think about introducing a `property` into `ResultBatch` that generates this dynamically?